### PR TITLE
feat: add config action

### PR DESCRIPTION
**Pull request by jbrockopp**
* Added a new `go-vela/cli/action/config` pkg
* Added a `Generate()` function to `go-vela/cli/action/config`
* Added tests for the `Generate()` function
* Added a `Remove()` function to `go-vela/cli/action/config`
* Added tests for the `Remove()` function
* Added a `Update()` function to `go-vela/cli/action/config`
* Added tests for the `Update()` function
* Added a `Validate()` function to `go-vela/cli/action/config`
* Added tests for the `Validate()` function
* Added a `View()` function to `go-vela/cli/action/config`
* Added tests for the `View()` function

### CODE REVIEW

File: action/config/doc.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+// Package config provides the defined CLI config actions for Vela.
+//
+// Usage:
+//
+// 	import "github.com/go-vela/cli/action/config"
+package config
```
Comment (jbrockopp) [inline action/config/doc.go:10]: Start of new `go-vela/cli/action/config` package

File: action/config/generate.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	"github.com/spf13/afero"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+// create filesystem based on the operating system
+var appFS = afero.NewOsFs()
+
+// Generate produces a pipeline based off the provided configuration.
+func (c *Config) Generate() error {
```
Comment (jbrockopp) [inline action/config/generate.go:19]: This is where we add the new `Generate()` function.

The purpose of this function is to generate a valid Vela configuration file for making requests.

File: action/config/generate.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	}
+
+	// create output for config file
+	out, err := yaml.Marshal(config)
+	if err != nil {
+		return err
+	}
+
+	// use custom filesystem which enables us to test
+	a := &afero.Afero{
```
Comment (jbrockopp) [inline action/config/generate.go:46]: We use the `github.com/spf13/afero` library so we can easily test by spinning up an in-memory filesystem.

File: action/config/generate_test.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+package config
+
+import (
+	"testing"
+
+	"github.com/spf13/afero"
+)
+
+func TestConfig_Config_Generate(t *testing.T) {
```
Comment (jbrockopp) [inline action/config/generate_test.go:13]: Here are the tests for the `Generate()` function

File: action/config/remove.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+	"github.com/go-vela/cli/internal/client"
+
+	"github.com/spf13/afero"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+// Remove deletes one or more fields from the config file based off the provided configuration.
+func (c *Config) Remove() error {
```
Comment (jbrockopp) [inline action/config/remove.go:18]: This is where we add the new `Remove()` function.

The purpose of this function is to remove one or more fields from the configuration file.

File: action/config/remove_test.go [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+package config
+
+import (
+	"testing"
+
+	"github.com/spf13/afero"
+)
+
+func TestConfig_Config_Remove(t *testing.T) {
```
Comment (jbrockopp) [inline action/config/remove_test.go:13]: Here are the tests for the `Remove()` function

File: action/config/update.go [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+	"github.com/go-vela/cli/internal/client"
+
+	"github.com/spf13/afero"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+// Update modifies one or more fields from the config file based off the provided configuration.
+func (c *Config) Update() error {
```
Comment (jbrockopp) [inline action/config/update.go:18]: This is where we add the new `Update()` function.

The purpose of this function is to modify one or more fields from the configuration file.

File: action/config/update_test.go [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+package config
+
+import (
+	"testing"
+
+	"github.com/spf13/afero"
+)
+
+func TestConfig_Config_Update(t *testing.T) {
```
Comment (jbrockopp) [inline action/config/update_test.go:13]: Here are the tests for the `Update()` function

File: action/config/validate.go [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+package config
+
+import (
+	"fmt"
+	"os"
+)
+
+// Validate verifies the configuration provided.
+func (c *Config) Validate() error {
```
Comment (jbrockopp) [inline action/config/validate.go:13]: This is where we add the new `Validate()` function.

The purpose of this function is to validate the configuration provided to the package.

File: action/config/validate_test.go [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package config
+
+import (
+	"testing"
+)
+
+func TestConfig_Config_Validate(t *testing.T) {
```
Comment (jbrockopp) [inline action/config/validate_test.go:11]: Here are the tests for the `Validate()` function

File: action/config/view.go [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+package config
+
+import (
+	"github.com/go-vela/cli/internal/output"
+
+	"github.com/spf13/afero"
+)
+
+// View inspects the config file based off the provided configuration.
+func (c *Config) View() error {
```
Comment (jbrockopp) [inline action/config/view.go:14]: This is where we add the new `View()` function.

The purpose of this function is to view a configuration file.

File: action/config/view_test.go [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+package config
+
+import (
+	"testing"
+
+	"github.com/spf13/afero"
+)
+
+func TestConfig_Config_View(t *testing.T) {
```
Comment (jbrockopp) [inline action/config/view_test.go:13]: Here are the tests for the `View()` function

File: action/config/generate.go (hunk 3 above)
Comment (davidvader) [reply action/config/generate.go:46]: very cool library

File: action/config/update.go [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+	// use custom filesystem which enables us to test
+	a := &afero.Afero{
+		Fs: appFS,
+	}
+
+	// send Filesystem call to read config file
+	data, err := a.ReadFile(c.File)
```
Comment (davidvader) [inline action/config/update.go:25]: If no config exists on their system does it error out here?

**Pull request merged**